shapeshift/web
# feat: cowswap tx parser

## Description



- [x] can parse cowswap txs
- [x] uses dynamic dex name for order route from parsed tx
- [x] dex names have been updated to user facing brand values, e.g. `thor` is now `THORChain`

TODO

- [x] requires unchained-client package bump to consume new parser from https://github.com/shapeshift/lib/pull/909
- [x] how to render fees @shapeshift/product - hidden for now if no fees on trade
<img width="1094" alt="Screen Shot 2022-07-25 at 6 48 38 PM" src="https://user-images.githubusercontent.com/88504456/180899019-153c1fa1-94e8-42bd-b17a-3f46ee4d9a38.png">


## Notice




## Pull Request Type


## Issue (if applicable)



closes #2218

## Risk



0x trades are interpreted as cowswap trades

## Testing



* enable cowswap flag
* do a cowswap trade (any erc20 -> erc20) will route via cowswap by default
* ensure that trade renders as per screenshot

## Screenshots (if applicable)

before

<img width="378" alt="image" src="https://user-images.githubusercontent.com/88504456/180898413-046387ce-fa50-4736-b72d-3ce431c170ce.png">

after

<img width="442" alt="image" src="https://user-images.githubusercontent.com/88504456/181074514-d7318890-2e87-4ed6-bad4-12e3f2d5feb0.png">
